sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonService.java

```diff
@@ -352,16 +362,24 @@ private Predicate createAssociationPredicate(@NotNull PersonQueryContext queryCo
 				eventParticipantService.createDefaultFilter(cb, joins.getEventParticipant()));
 			break;
 		case IMMUNIZATION:
-			associationPredicate = CriteriaBuilderHelper.and(
-				cb,
-				immunizationService.createUserFilter(new ImmunizationQueryContext(cb, cq, joins.getImmunizationJoins())),
-				immunizationService.createDefaultFilter(cb, joins.getImmunization()));
+			if (!isRestrictedToAssignedEntities()) {
+				associationPredicate = CriteriaBuilderHelper.and(
+					cb,
+					immunizationService.createUserFilter(new ImmunizationQueryContext(cb, cq, joins.getImmunizationJoins())),
+					immunizationService.createDefaultFilter(cb, joins.getImmunization()));
+			} else {
+				associationPredicate = CriteriaBuilderHelper.and(cb, cb.disjunction());
```

Comment:
shpild this really be `disjunction()`, doesn't it mean you won't see any immmunization related persons?

Reply:
removed the disjunction